components/src/core/components/Input/Select/SelectInput.vue

```diff
@@ -187,6 +195,125 @@ export default defineComponent({
+
+          const referenceIndex = this.computedOptions.findIndex(
+            (option: Option) => option.id === newId,
+          );
+          if (
+            referenceIndex !== -1 &&
+            referenceIndex < this.computedOptions.length - 1
+          ) {
+            // Get the next option and auto-select it
+            const nextOption = this.computedOptions[referenceIndex + 1];
+            if (nextOption && !nextOption._disabled) {
+              this.isAutoSelected = true; // Mark as auto-selected
+
+              // Reset pointer to ensure inputValue updates correctly
+              this.pointer = -1;
+
+              // Force update by using nextTick
+              this.$nextTick(() => {
+                this.$emit('update:modelValue', nextOption);
+              });
+            }
+          }
+        }
+
+        if (!newValue) {
+          // Reset tracking when scrollToNextOf is cleared
+          this.lastScrollToNextOfId = null;
+          this.isAutoSelected = false;
+        }
+      },
+      immediate: true,
+      deep: true,
+    },
+    dropdownOpen(isOpen: boolean) {
+      if (isOpen) {
+        // Dropdown is OPENING - reset interaction flag
+        this.userInteractedWithDropdown = false;
+        this.valueWhenDropdownOpened = this.modelValue?.id || null;
+
+        let targetIndex = -1;
+
+        // Priority 1: If there's a selected value, scroll to it
+        if (this.modelValue) {
+          targetIndex = this.computedOptions.findIndex(
+            (option: Option) => option.id === this.modelValue.id,
+          );
+        }
+        // Priority 2: If no selection but scrollToNextOf is provided, scroll to next after reference
+        else if (this.scrollToNextOf) {
+          const referenceIndex = this.computedOptions.findIndex(
+            (option: Option) => option.id === this.scrollToNextOf.id,
+          );
+          if (
+            referenceIndex !== -1 &&
+            referenceIndex < this.computedOptions.length - 1
+          ) {
+            // Get the next option index
+            targetIndex = referenceIndex + 1;
+          }
+        }
+
+        // Scroll to the target index if found
+        if (targetIndex !== -1) {
+          this.pointer = targetIndex;
+
+          this.$nextTick(() => {
+            const optionRef = this.$refs[`option-${targetIndex}`];
+            // Handle both array and single ref cases
+            const optionElement = Array.isArray(optionRef)
+              ? optionRef[0]
+              : optionRef;
+            if (optionElement?.$el) {
+              // Find the dropdown inner container
+              const dropdownInner = optionElement.$el.closest(
+                '.oxd-select-dropdown-inner',
+              );
+              if (dropdownInner) {
+                const optionTop = optionElement.$el.offsetTop;
+                // Position target item at the top of the dropdown
+                dropdownInner.scrollTop = optionTop;
```

Comment:
Better to use [scrollIntoView](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) native method